src/libraries/System.Memory.Data/tests/BinaryDataTests.cs

```diff
@@ -431,6 +431,59 @@ public void MaxStreamLengthRespected()
             var data = BinaryData.FromStream(new OverFlowStream(offset: int.MaxValue - 1000));
         }
 
+        [Fact]
+        public async Task CanCreateBinaryDataFromFile()
+        {
+            byte[] buffer = "some data"u8.ToArray();
+            string path = Path.GetTempFileName();
+            File.WriteAllBytes(path, buffer);
```

Comment:
Should temp files be cleaned up?

Reply:
Indeed!
Also fixed for the existing test that I took as reference.